User/InMemoryUserRepository.php

```diff
@@ -50,22 +50,22 @@ public function getSupportedClass()
     /**
      * {@inheritdoc}
      */
-    public function add(User $user)
+    public function add(UserInterface $user)
```

Comment:
waarom verander je dat hier? Je wil in één repository toch niet users van verschillende classes kunnen opslaan? Normaal sla je altijd zaken van hetzelfde type op in een repository.